daprdocs/content/en/developing-applications/building-blocks/state-management/query-state-store-api.md

```diff
@@ -0,0 +1,265 @@
+---
+type: docs
+title: "State Query API"
+linkTitle: "State Query API"
+weight: 600
+description: "API for querying state stores"
+---
+
+## Introduction
+
+The query API provides unified way of querying state stores regardless of underlying databases.
+
+{{% alert title="In preview" color="warning" %}}
+The state store query API is currently in [preview]({{< ref preview-features.md >}}). To enable, opt-in to `State.Query` feature.
+{{% /alert %}}
+{{% alert title="Note" color="primary" %}}
+The state store query API is implemented per component. Currently we support `MongoDB` and `Azure CosmosDB`. Support for more components is underway.
+{{% /alert %}}
+
+## Syntax of the query request
+
+Users submit query requests via HTTP POST or gRPC.
+The body of the request is the JSON map with 3 entries: `filter`, `sort`, and `pagination`
+The `filter` section specifies the query conditions in the form of a tree of key/values operations, where the key is the operator and the value is the operands.
+
+Currently we support four operations:
+| Operator | Operands | Description |
+|----------|-----------|-------------|
+| `EQ`       | key:value  | key == value|
+| `IN`        | key:[]value| key == value[0] OR key == value[1] OR ... OR key == value[n] |
+| `AND`    | []operation|  operation[0] AND operation[1] AND ... AND operation[n] |
+| `OR`      | []operation|  operation[0] OR operation[1] OR ... OR operation[n] |
+
+The `sort` is an optional section is an ordered array of `key:order` pairs, where `key` is a key in the state store, and the `order` is an optional string indicating sorting order: `"ASC"` for ascending and `"DESC"` for descending. If omitted, ascending order is assumed.
+
+The `pagination` is an optional section containing `limit` and `token` parameters. `limit` sets the page size. `token` is an iteration token returned by the component, and used in consecutive queries.
+
+Here is an example of the query request JSON body:
+```json
+{
+    "query": {
+        "filter": {
+            "OR": [
+                {
+                    "EQ": {"person.org": "Dev Ops"}
+                },
+                {
+                    "AND": [
+                        {
+                            "EQ": {"person.org": "Finance"}
+                        },
+                        {
+                            "IN": {"state": ["CA", "WA"]}
+                        }
+                    ]
+                }
+            ]
+        },
+        "sort": [
+            {
+               "key": "state",
+                "order": "DESC"
+            },
+            {
+                "key": "person.name"
+            }
+        ],
+        "pagination": {
+            "limit": 3
+        }
+    }
+}
+```
+
+The query request will be eventually translated into a native query language and executed in the state store component.
```

Comment:
Use present tense. 
"The query request is translated into a native query language and executed in the state store component."